src/Compilers/CSharp/Portable/Binder/Binder_Deconstruct.cs

```diff
@@ -663,7 +693,18 @@ private static void FlattenDeconstructVariables(ArrayBuilder<DeconstructionVaria
                 case SyntaxKind.DeclarationExpression:
                     {
                         var component = (DeclarationExpressionSyntax)node;
-                        return BindDeconstructionDeclarationVariables(component.Type, component.Designation, diagnostics);
+                        bool isVar;
+                        bool isConst = false;
+                        AliasSymbol alias;
+                        TypeSymbol declType = BindVariableType(component.Designation, diagnostics, component.Type, ref isConst, out isVar, out alias);
+                        Debug.Assert(isVar == ((object)declType == null));
+                        if (component.Designation.Kind() == SyntaxKind.ParenthesizedVariableDesignation && !isVar)
+                        {
+                            // An explicit is not allowed with a parenthesized designation
+                            Error(diagnostics, ErrorCode.ERR_DeconstructionVarFormDisallowsSpecificType, component.Designation);
```

Comment:
In the original code, a bound node would be marked with `hasErrors`. Is this difference significant?